src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -114,8 +116,16 @@ export default function connectHierarchicalMenu(renderFn, unmountFn = noop) {
       },
 
       init({ helper, createURL, instantSearchInstance }) {
+        sendEvent = createSendEventForFacet({
+          instantSearchInstance,
+          helper,
+          attribute: hierarchicalFacetName,
+          widgetType: 'ais.refinementList',
+        });
+
         this.cachedToggleShowMore = this.cachedToggleShowMore.bind(this);
         this._refine = function(facetValue) {
+          sendEvent('click', facetValue);
```

Comment:
doesn't check if it's refined already

Reply:
if the check is in sendEvent, why don't we pass `isRefined` too, instead of recomputing

Reply:
`createSendEventForFacet` checks.

`isRefined` is not given to `_refine` and I guess that's sort of a reason why `helper.toggleRefinement()` doesn't get it either and checks internally.

Indeed, there's a re-computation going on in `sendEvent`. Not sure how to deduplicate it without changing too much of helper.